ch_pipeline/processing/daily.py

```diff
@@ -407,29 +410,136 @@ def _load_hook(self):
+            ]
+
         self._padding = self._revparams["padding"]
 
+    def _available_files(self, start_csd, end_csd):
+        """
+        Return chimestack files available in cedar_online between start_csd and end_csd, if all of the files for that period are available online.
+
+        Return an empty list if files between start_csd and end_csd are only partially available online.
+
+        Total file count is verified by checking cedar_nearline.
+
+        Parameters
+        ----------
+        start_csd : int
+            Start date in sidereal day format
+        end_csd : int
+            End date in sidereal day format
+
+        Returns
+        -------
+        list
+            List contains the chimestack files available in the timespan, if all of them are available online
+
+        """
+
+        # Connect to databases
+        db.connect()
+
+        # Get timestamps in unix format
+        # Needed for queries
+        start_time = ephemeris.csd_to_unix(start_csd)
+        end_time = ephemeris.csd_to_unix(end_csd)
+
+        # We will want to know which files are in chime_online and nearline on cedar
+        online_node = [di.StorageNode.get(name="cedar_online", active=True)]
+        nearline_node = [di.StorageNode.get(name="cedar_nearline", active=True)]
+
+        # TODO if the time range is so small that it’s completely contained within a single file, nothing will be returned
+        # have to special-case it by looking for files which start before the start time and end after the end time).
+
+        archive_files = (
+            di.ArchiveFileCopy.select(
+                di.ArchiveFileCopy.node,
+                di.ArchiveFile,
+                di.ArchiveAcq,
+                di.ArchiveInst,
+                di.CorrFileInfo,
+            )
+            .join(di.ArchiveFile)
+            .join(di.ArchiveAcq)
+            .join(di.ArchiveInst)
+            .switch(di.ArchiveFile)
+            .join(di.CorrFileInfo)
+        )
+
+        # chimestack files available online which include between start and end_time
+        files_online = archive_files.where(
+            di.ArchiveInst.name
+            == "chimestack",  # specifically looking for chimestack fiels
+            di.CorrFileInfo.start_time
+            < end_time,  # which contain data that includes start time and end time
+            di.CorrFileInfo.finish_time >= start_time,
+            di.ArchiveFileCopy.node == online_node,  # that are online
+            di.ArchiveFileCopy.has_file == "Y",
+        )
+
+        # chimestack files available on nearline, which include between start and end_time
+        files_nearline = archive_files.where(
+            di.ArchiveInst.name == "chimestack",
+            di.CorrFileInfo.start_time < end_time,
+            di.CorrFileInfo.finish_time >= start_time,
+            di.ArchiveFileCopy.node == nearline_node,  # that are on nearline
+            di.ArchiveFileCopy.has_file == "Y",
```

Comment:
The second suggestion above, with a bit of simplification to make it clearer that one set is a subset of the other
```suggestion
        # chimestack files available anywhere between start and end_time
        files_that_exist = archive_files.where(
            di.ArchiveInst.name
            == "chimestack",  # specifically looking for chimestack fiels
            di.CorrFileInfo.start_time
            < end_time,  # which contain data that includes start time and end time
            di.CorrFileInfo.finish_time >= start_time,
            di.ArchiveFileCopy.has_file == "Y",
        )

        # chimestack files available on online
        files_online = files_that_exist.where(
            di.ArchiveFileCopy.node == online_node,  # that are on online
        )
```